### PR TITLE
Make documentation more user-friendly

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

Ralph's documentation is high-level techy-oriented but it should be more user-centric without any a pre-requisite on user knowledge.

## Proposal

- [x] make the project README more useful
- [x] clarify the many-faces aspects of the project
- [x] improve the getting started section

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -136,10 +149,10 @@ down: ## stop and remove backend containers
 	@$(COMPOSE) down
 .PHONY: down
 
-es-index: ## create elasticsearch index and sample documents
+es-index: ## create elasticsearch index
 es-index: run-es
 	@echo "Creating $(ES_INDEX) index"
-	bin/es index $(ES_INDEX)
+	curl -X PUT $(ES_URL)/$(ES_INDEX)
```
Comment (SergioSim) [inline Makefile:155]:
```suggestion
	curl -X PUT $(ES_URL)/$(ES_INDEX)
	curl -X PUT ${ES_URL}/$(ES_INDEX)/_settings -H 'Content-Type: application/json' -d '{"index": {"number_of_replicas": 0}}'
```
I really like your solution)
Could we also set the number of replicas to 0? It seems that ES refuses to exit the "yellow" state when it's unable to meet the default replication factor.

File: README.md [hunk 2]
```diff
@@ -3,15 +3,109 @@
...
+### Testing the LRS server with Docker compose
+
+> [`curl`](https://curl.se) and [`jq`](https://stedolan.github.io/jq/) are
+> required to run some commands of this tutorial.
+
+To bootstrap a test environment on your machine, clone this project first and
+run the `bootstrap` Makefile target:
+
+```
+$ make bootstrap
```
Comment (SergioSim) [inline README.md:25]: As `make bootstrap` starts the Elasticsearch container that is used afterwards, could we add the note about the `virtual memory limits` from https://github.com/openfun/ralph/blob/master/docs/contribute.md?plain=1#L72-L80 ?

File: README.md [hunk 3]
```diff
@@ -3,15 +3,109 @@
...
+
+```
+$ curl -s \
+    --user ralph:secret \
+    -H "Content-Type: application/json" \
+    http://localhost:8100/xAPI/statements/ \ |
+  jq
+```
+
+> Note that using `jq` is optionnal in this case, it is used to improve
```
Comment (SergioSim) [inline README.md:92]:
```suggestion
> Note that using `jq` is optional in this case, it is used to improve
```

**Review by SergioSim — COMMENTED** [review]
This looks great)
Have only minor questions/suggestions)
image

File: .env.dist [hunk 4]
```diff
@@ -35,21 +35,21 @@ RALPH_APP_DIR=/app/.ralph
...
-# RALPH_BACKENDS__DATABASE__MONGO__COLLECTION=foo
-# RALPH_BACKENDS__DATABASE__MONGO__DATABASE=statements
-# RALPH_BACKENDS__DATABASE__MONGO__CONNECTION_URI=mongodb://localhost:27017/
-# RALPH_BACKENDS__DATABASE__MONGO__TEST_COLLECTION=foo
-# RALPH_BACKENDS__DATABASE__MONGO__TEST_FORWARDING_COLLECTION=foo-2
-# RALPH_BACKENDS__DATABASE__MONGO__TEST_DATABASE=statements
-# RALPH_BACKENDS__DATABASE__MONGO__TEST_CONNECTION_URI=mongodb://mongodb:27017/
+RALPH_BACKENDS__DATABASE__MONGO__COLLECTION=foo
+RALPH_BACKENDS__DATABASE__MONGO__DATABASE=statements
+RALPH_BACKENDS__DATABASE__MONGO__CONNECTION_URI=mongodb://localhost:27017/
```
[the remark this answers is not in the document]
Comment (jmaupetit) [reply .env.dist:48]: While the `.env` file is designed to be used in a development environment, defaults in Ralph's code should be more generic (host names in connection URLs are linked to our docker-compose stack...).

[the remark this answers is not in the document]
Comment (jmaupetit) [reply .env.dist:48]: But I've fixed mongodb's connection URL. Thx!

File: Makefile (hunk 1 above)
Comment (jmaupetit) [reply Makefile:155]: Good point!

File: README.md [hunk 5]
```diff
@@ -3,15 +3,109 @@
...
+    [xAPI](https://adlnet.gov/projects/xapi/), or
+    [openedx](https://docs.openedx.org/en/latest/developers/references/internal_data_formats/tracking_logs/index.html)
 - **command-line interface** (CLI), to build data pipelines the UNIX-way™️,
 - **HTTP API server**, to collect xAPI statements (learning events)
   following the [ADL LRS
   standard](https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Communication.md#partthree).
 
 ## Quick start guide
 
+### Testing the LRS server with Docker compose
```
Comment (wilbrdt) [inline README.md:16]: I think a small prerequisite for docker compose would be nice, WDYT? On what version is needed and maybe a link to know how to install it?

File: README.md (hunk 2 above)
Comment (jmaupetit) [reply README.md:25]: Good point. I've fixed that!

File: README.md (hunk 5 above)
Comment (jmaupetit) [reply README.md:16]: Good point. I've added preliminary notes for this.

**Review by SergioSim — APPROVED** [review]
LGTM)

image

File: README.md [hunk 6]
```diff
@@ -3,15 +3,121 @@
...
+
+```
+$ curl -s \
+    --user ralph:secret \
+    -H "Content-Type: application/json" \
+    http://localhost:8100/xAPI/statements/ \ |
+  jq
+```
+
+> Note that using `jq` is optional in this case, it is used to improve response
```
Comment (quitterie-lcs) [inline README.md:104]: Maybe also add a remark that jq may not be installed on user computer

File: README.md [hunk 7]
```diff
@@ -3,15 +3,121 @@
...
+  sed "s/@timestamp/timestamp/g" | \
+  jq -s . | \
+  curl -Lk \
+    --user ralph:secret \
+    -X POST \
+    -H "Content-Type: application/json" \
+    http://localhost:8100/xAPI/statements/ -d @-
+```
+
+The command above fetches 100 (one hundred) example xAPI statements from our
```
Comment (quitterie-lcs) [inline README.md:89]: Why are you writing 100 in words between parenthesis?

File: README.md [hunk 8]
```diff
@@ -3,15 +3,121 @@
...
+
+```json
+{"database":"ok"}
+```
+
+If the database status is satisfying, you are now ready to send xAPI statements
+to the LRS:
+
+```
+$ curl -sL https://github.com/openfun/potsie/raw/main/fixtures/elasticsearch/lrs.json.gz | \
```
Comment (quitterie-lcs) [inline README.md:77]: Maybe in a next PR have an archive in ralph repo? WDYT?

**Review by wilbrdt — APPROVED** [review]
OK for me! 

Alt Text

File: README.md (hunk 8 above)
Comment (jmaupetit) [reply README.md:77]: Or in a dedicated repository to avoid duplicating the same data set again and again?

File: README.md (hunk 7 above)
Comment (jmaupetit) [reply README.md:89]: It's a matter of style, but I need to reverse it as:

```suggestion
The command above fetches one hundred (100) example xAPI statements from our
```

**Pull request merged**